src/jumbo.c

```diff
@@ -300,6 +300,19 @@ char *strlwr(char *s)
 }
 #endif
 
+void memcpylwr(char *dest, const char *src, size_t n)
+{
+	while (n--) {
+		if (*src >= 'A' && *src <= 'Z') {
+			*dest = *src | 0x20;
```

Comment:
I just copied that part of the logic from `strlwr` so if that worked before, this will do the exact same thing.

Reply:
Every single encoding we support (and will ever support) is a superset of ASCII so that's not a problem

Reply:
We support EBCDIC but only a very limited set (the SAP for sure).  We do not handle that with our encoding logic at all (even though we 'could'.)

Reply:
Now that I'm less tired: This obviously will not lowercase the non-ASCII parts of any encoding we support (obviously what @jfoug meant). As long as it's only used in eg. split (and with care) it should be fine. It can never replace `enc_strlwr()`. MSCASH format needs (and has) the latter because ciphertext may contain a username with non-ASCII.

Reply:
ding-ding!  If used judiciously (just like enc_strlwr() should be), and ASCII only (str)|(mem)cpy with casing is just fine.  a LARGE part of what john does, IS just ASCII only.   

But that is a caveat when coding for john. One DOES need to know the few nuanced places where ASCII only is not acceptable to be used.  

I do think this REALLY, is a set of functions designed for usage in split/prepare to get the source strings into a unified layout.  Its usage should be limited to there.  Once it starts getting used all over the place, people will become comfortable using it, and it will get used in wrong places. Then, unless --encoding=xxx is used, it can be a bug that stays dormant and in production (bleeding) code obtainable by users for quite some time.

Speaking of such, do we use any -enc=xxx testing in our CI's to make sure we do not allow encode errors to linger in checked in code?

Reply:
Of course, when I suggested this I meant it only for use on hash encodings, not on password candidates.

Reply:
Yes but eg. MSCASH has things (like username, domain name) that may be non-ASCII within the ciphertext

Reply:
Just rename that function so that it has ascii in the name. And make sure MSCASH uses a separate function.

Reply:
We have strlwr() and enc_strlwr(). I think this name is OK.